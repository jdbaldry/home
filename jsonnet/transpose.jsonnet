local util = import 'util.libsonnet';

local not_array = 0;
local empty_array = [];
local simple_array = [1, 2, 3];
local two_dimensional = [[1, 2, 3], [4, 5, 6], [7, 8, 9]];
local mismatched_lengths = [[1, 2], [4, 5, 6], [7, 8], [10, 11, 12]];
{
    // RUNTIME ERROR: argument must be an array
    // not: util.transpose(not_array)
    test_empty: util.transpose(empty_array),
    // TODO: should this be null or [ ],
    test_simple: util.transpose(simple_array),
    test_two_dimensional: util.transpose(two_dimensional),
    test_mismatched_lengths: util.transpose(mismatched_lengths),
}  