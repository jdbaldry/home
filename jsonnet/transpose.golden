{
   "test_empty": [ ],
   "test_mismatched_lengths": [
      [
         1,
         4,
         7,
         10
      ],
      [
         2,
         5,
         8,
         11
      ],
      [
         6,
         12
      ]
   ],
   "test_simple": null,
   "test_two_dimensional": [
      [
         1,
         4,
         7
      ],
      [
         2,
         5,
         8
      ],
      [
         3,
         6,
         9
      ]
   ]
}